shapely/geos.py

```diff
@@ -555,7 +553,32 @@ def errcheck_predicate(result, func, argtuple):
     return result
 
 
-class LGEOSBase(threading.local):
+def _get_lgeos_base():
+    """Get the class that should be used as a base class for LGEOSBase.
+
+    Gevent will replace `threading.local` with an implementation that makes all
+    thread local variables greenlet local instead.  In the context of shapely,
+    this means all objects that inherit from LGEOSBase will become greenlet
+    local instead of thread local. Effectively, this causes servers that
+    utilize both gevent and shapely to re-initialize the geos bindings every
+    request.
+
+    Since LGEOSBase inherits from `threading.local` to prevent concurrent
+    access to the GEOS C API, this is not relevant in a monkey patched gevent
+    process which will only ever run one thread at a time.  In this case, we
+    inherit from `object` instead of `threading.local`
+    """
+    result = threading.local
+    try:
+        from gevent import monkey
+        if 'threading' in monkey.saved:
+            result = object
+    except Exception:
```

Comment:
@rowillia how about catching the most specific error(s) possible? `ImportError` or whatever.

Reply:
@rowillia just following up, I've got this one change request blocking merge of the PR.